flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/windowing/TopSpeedWindowing.java

```diff
@@ -99,7 +104,20 @@ public static void main(String[] args) throws Exception {
                             .map(new ParseCarData())
                             .name("parse-input");
         } else {
-            carData = env.addSource(CarSource.create(2)).name("in-memory-source");
+            CarGeneratorFunction carGenerator = new CarGeneratorFunction(2);
+            DataGeneratorSource<Tuple4<Integer, Integer, Double, Long>> carGeneratorSource =
+                    new DataGeneratorSource<>(
+                            carGenerator,
+                            Long.MAX_VALUE,
+                            parallelismIgnored -> new GuavaRateLimiter(1),
+                            TypeInformation.of(
+                                    new TypeHint<Tuple4<Integer, Integer, Double, Long>>() {}));
+            carData =
+                    env.fromSource(
+                            carGeneratorSource,
+                            WatermarkStrategy.noWatermarks(),
```

Comment:
Yes, and the watermarks are assigned here https://github.com/apache/flink/pull/21774/files#diff-227d83380c24406389c12d4d3116ced03c15493d638b029595431051695c2549L109 because of this if-else block
https://github.com/apache/flink/pull/21774/files#diff-227d83380c24406389c12d4d3116ced03c15493d638b029595431051695c2549L86 and the file source which parses data in a separate map function. There is some consensus that this example does not really serve its purpose, so I don't think it makes sense to put too much effort into "fixing" it: https://github.com/apache/flink/pull/20049#issuecomment-1407553971

Reply:
> There is consensus that this example does not really serve its purpose, so I don't think it makes sense to put much effort into "fixing" it, it is enough that it produces data and compiles.

As for this `TopSpeedWindowing` example, it is fair enough as I'm also not a fan of this. Perhaps we should show users a more intuitive example that cover the core logic of this in the future.

Reply:
To be honest I am not quite sure which feature this example originally intended to cover. I know that there is also some consensus that usage of advanced windows with triggers and evictors is somewhat discouraged and should not be promoted. 
cc @alpinegizmo would appreciate your opinion

Reply:
> There is also some consensus that usage of advanced windows with triggers and evictors is somewhat discouraged and should not be promoted.
Agree on this point.  Whether removing or improving this example, it should not block this migrate ATM.